app/clients/salesforce/salesforce_client.py

```diff
@@ -98,5 +98,22 @@ def engagement_update_stage(self, service: Service, user: User, stage_name: str)
         """
         session = self.get_session()
         account_id, contact_id = self.contact_update_account_id(session, service, user)
-        salesforce_engagement.update_stage(session, service, stage_name, account_id, contact_id)
+        stage_update = {"StageName": stage_name}
+        salesforce_engagement.update(session, service, stage_update, account_id, contact_id)
+        self.end_session(session)
+
+    def engagement_close(self, service: Service):
+        """Closes a Salesforce Engagement for the given Notify service.
+
+        Args:
+            service (Service): Notify Service to close an Engagement for.
+        """
+        session = self.get_session()
+        engagement = salesforce_engagement.get_engagement_by_service_id(session, str(service.id))
+        if engagement:
+            close_update = {
+                "CDS_Close_Reason__c": "Trial deleted by user",
```

Comment:
This could also happen to a live account, correct?
```suggestion
                "CDS_Close_Reason__c": "Service deleted by user",
```